runtime/doc/lua.txt

```diff
@@ -875,6 +875,19 @@ vim.str_byteindex({str}, {index}[, {use_utf16}])	*vim.str_byteindex()*
         in the middle of a UTF-16 sequence is rounded upwards to the end of that
         sequence.
 
+vim.iconv({str}, {from}, {to})                                  *vim.iconv()*
+        The result is a String, which is the text {str} converted
+        from encoding {from} to encoding {to}.
+        When the conversion completely fails an empty string is
+        returned.  When some characters could not be converted they
+        are replaced with "?".
+        The encoding names are whatever the iconv() library function
+        can accept, see ":!man 3 iconv".
+        Note that Neovim uses UTF-8 for all Unicode encodings, conversion
+        from/to UCS-2 is automatically changed to use UTF-8.  You
+        cannot use UCS-2 in a string anyway, because of the NUL bytes.
```

Comment:
well this is not true with lua strings. a lua string can be UTF-16 or 32 or whatever it wants to. we should lift restrictions like this if we add this function.

Reply:
I'm not sure what part of the code introduces this restriction. My guess is that it is done by `convert_setup` which is an alias for `convert_setup_ext` with `from_unicode_is_utf8=true` and `from_unicode_is_utf8=true`